Add macOS MDM profiles and Windows support.

williamtheaker opened the pull request:
We forked this cookbook at some point and have drifted quite a bit, so I'm submitting our changes upstream.

This PR adds Windows support by using Zoom's [registry keys](https://support.zoom.us/hc/en-us/articles/360039100051), so we can configure the client on-the-fly rather than at install/upgrade.

Tested by running on a fresh Zoom install, Zoom installed via Chef with Chocolatey/MSI flags, and standalone Zoom without install-time flags. In my testing, registry keys take precedence over MSI flags, which allows pushing this out to existing install-bases.

Applying new settings via the registry merely requires quitting and reopening Zoom.

ChefAustin commented:
Thanks for contributing, @williamtheaker !

I'll try to review this by EOW.

ChefAustin commented on chef/cookbooks/cpe_zoom/resources/cpe_zoom.rb:
```diff
@@ -17,21 +15,110 @@
+
+      preference = WINDOWS_CHAT_SETTINGS_MAP[key]
+      if preference.nil? # If the preference isn't in WINDOWS_CHAT_SETTINGS_MAP, set General path.
+        zoom_settings[WINDOWS_GENERAL_SETTINGS_MAP.fetch(key, key)] = { "path": 'HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Zoom\Zoom Meetings\General', "name": WINDOWS_GENERAL_SETTINGS_MAP.fetch(key, key) }.merge(convert_to_registry_key(value))
```
Have you run Rubocop against this source feature branch (using the `.ruboop.yml` from the root of our project repo)? 

It looks like this line is ~240chars (and our max `LineLength` for Rubocop is 120chars).

williamtheaker replied:
done

williamtheaker commented:
I updated and cleaned up `.rubocop.yml`, since using it with a modern version of `cookstyle` fails with: `(obsolete configuration found in .rubocop.yml, please update it)`.
